### ISSUE TITLE
3.3.7 Accessible Authentication disallows Security questions

### ISSUE THREAD

**Issue by mbgower** [w3c/wcag#2733]
The fourth bullet of [3.3.7](https://www.w3.org/TR/WCAG22/#accessible-authentication) on allowable personal content is restricted to a test that identifies “non-text content the user provided to the website.”

There is nothing in the Understanding document that explains why _text_ provided by the user is excluded from allowable personal content. Does the recognition of a user-supplied image place less demand on memory? Can the rationale please be explained?

We have a concern that text-based security questions form a _very_ common part of many authentication/reauthentication processes. Such mechanisms are typically established during account set-up. The user is provided with a list of preset questions to choose from (e.g., “What is your oldest sibling’s middle name?) and asked to populate the correct answers for a certain number. The security questions are then used as part of a login (or more often password reset) process. The user is presented with one (or more) of their chosen questions, and asked to provide the same answer.

Allowing the user to provide an answer to something easier for them to remember is a very common part of re-authentication. By specifically excluding user-supplied text, the Success Criterion eliminates the use of a common safeguard against identify theft, and probably results in a failure of this Criterion for a significant number of existing mechanisms.

**Comment by mbgower** [comment]
> Examples of mechanisms that satisfy this criterion include:
1. support for password entry by password managers to reduce memory need, and

Also, the same password manager that stores passwords to reduce memory need can also store the answers to these questions, so it does not seem to create significant additional effort, when such a prompt is necessary.

**Comment by bruce-usab** [comment]
Thanks for highlighting this Mike.  My own thinking was that these sort of security question fall into the category of "Another authentication method that does not rely on a cognitive function test."  Even if that is correct, it would be better for that to be explicit in the SC or as a Note.

But if those sort of security questions are not a cognitive function test, then neither is a cognitive function test for the end-user recognizing an object they previously uploaded to a website.

**Comment by charleshalldesign** [comment]
This type of security question is indeed a cognitive function test. It specifically relies on memory to create the original answer and to recall it again later. Memory should not be used as a proxy for security. The prevalence of this pattern does not make it usable, accessible, acceptable, or secure. 

My interpretation of the SC is that it does not disallow or prevent this type of security questions. Rather, it better informs them that an accessible alternate method such as selecting from possible answers should be included as part of the pattern.

Example that would satisfy “A mechanism is available to assist the user in completing the cognitive function test.”:

Original input
Q: What is your oldest sibling’s middle name?
A: Jane

Subsequent requested input for confirmation
Q: What is your oldest sibling’s middle name?
A: Select from:

1. Jade
2. Jane
3. June

Note: this still relies on memory –  a cognitive function test – to fulfill the original input and the recall the value again. However, it satisfies the criterion and is regarded as assisting memory.

**Comment by patrickhlauke** [comment]
> Rather, it better informs them that an accessible alternate method such as selecting from possible answers should be included as part of the pattern.

lowercase (as in not normative) "should", right?

**Comment by charleshalldesign** [comment]
Regarding the original questions:
“Does the recognition of a user-supplied image place less demand on memory? Can the rationale please be explained?”

Yes. In general, this follows the principle of recognition over recall. It requires less effort and less reliance on memory to recognize text or an object in the current context than to recall it from a previous one. Importantly, text and images are often mapped differently in memory, as this gets into sensory memory. However, presenting text back on a screen becomes a hybrid of different memory methods for a sighted user. Seeing a word can trigger different memories, thus increasing the chance that one will be accurate.

**Comment by GreggVan** [comment]
No only less burden on memory - but also on history and information exposure.    If they put the image up  - they will know what it is. 

I had a capcha the other day that said I was to point out all the  (somethings)  and  I had no idea what the something was.  

People with cognitive disabilities may have less broad experiences to draw from.    If the put it up… 

gregg

------------------------------
Gregg Vanderheiden
***@***.***

**Comment by alastc** [comment]
> There is nothing in the Understanding document that explains why text provided by the user is excluded from allowable personal content. Does the recognition of a user-supplied image place less demand on memory? Can the rationale please be explained?

I think your description of the usage helped to explain, e.g. "The user is provided with a list of preset questions to choose from (e.g., “What is your oldest sibling’s middle name?) and asked to populate the correct answers".

When it is text based the standard usage is to display the question, and the user has to remember (recall) and transcribe (type in) the answer(s).

When it is image pages, the standard(ish, it isn't common) usage is to display a set of images and the user has to recognise (not recall) the image, and select it.

Recognition vs recall is [well known to create big differences in usability](https://www.nngroup.com/articles/recognition-and-recall/), and whilst we must acknowledge is still a barrier, for most people with cognitive issues it is a smaller barrier than text recall & transcription. 

I agree with @charleshalldesign that providing multiple choice could be a 'mechanism to assist', although I wouldn't want to promote that as a technique. It would be better to avoid it altogether. 


> By specifically excluding user-supplied text, the Success Criterion eliminates the use of a common safeguard against identify theft, and probably results in a failure of this Criterion for a significant number of existing mechanisms.

I agree a lot of places use this, but I also have some very secure accounts with organisations that do not use that approach. I haven't looked it up, but I'd bet $10 that national security agencies don't recommend the method as it's relatively easy to research someone and find enough answers to impersonate someone.

Generally these are mechanisms to _get around_ the authentication process (e.g. because you've lost your password, or changed email address or something), rather than to authenticate. There are other methods, such as email round-trips, or phoning / text-chatting with customer services (who may ask you such questions outside the authentication process).

For clarification, I've added this to another branch where the whole does it a bit better structured:
"Text-based personal content does not qualify for this exception as it relies on recall (rather than recognition), and transcription (rather than selecting an item). Whilst picture-based personal content will still be a barrier for some people, text based versions tend to be a much larger barrier."

Line 97 in #2811

**Comment by alastc** [comment]
> The proposed replacement (choosing which object has been uploaded) 

That isn't exactly a proposed replacement, it is something that has an exception.

The proposed replacement (considering security) would be something that isn't part of the normal authentication, e.g. an email loop or something you have.

If it is coming across as something we propose, that should be something else we update the understanding doc about.
